### ISSUE TITLE
not support odbc so not usefully

### ISSUE THREAD

**Issue by mckaygerhard** [Deganius/gb.deg.form#2]
the hole component depends on how grid are filled, due some odbc implementations does are `Foward only` with the cursor that return to the odbc, and the gambas module only implements that in the odbc can able to do..

so 
> Some components for displaying data of lists and results of database-queries.
are totally false, it only works with mysql and partially with sqlite

so in the Gridview.class i can see:

```
    If $res.Count > 0 Then
```

that are complety incompatible with odbc due as that bug in gambas said:
http://gambaswiki.org/bugtracker/edit?object=BUG.1100&from=L21haW4-
odbc standar doe snot return count.. 

about sqlite, partially support due some problems with behavioour of the sqlite, as said:
> If there is no AS clause then the name of the column is unspecified and may change from one release of SQLite to the next.

As well as many other things not specified here, when a query is not specific the behavior between versions of sqlite is unexpected .. because there are some settings (which I already did) that need to be done, but I still limit it to two databases since With postgres not working either complety--

In short if it can not be used in any database, it is a component that can only be a library, since it is not independent of the database as other components are

**Comment by chthal** [comment]
Thanks. It was written for Postgresql, there (and AFAIK with Mysql) it works flawlessly. And it was made as a component to provide drag and drop inside the form designer of the Gambas IDE. It is not designed to  be a generic Gambas component.

**Comment by mckaygerhard** [comment]
i just tested with postgres, some querys get a segfaul, i cannot give you specific querys but i can see more in the code the next week and propose something...  i made with help of some gambas users from gambas.-es a better component but only works with sqlite.. i right now make it some newer to work with odbc

**Comment by chthal** [comment]
If you cannot provide an example for a query, then your posting does not help in any way. gb.deg.form now works with Postgresql and Mysql in production code every day for more than two years.

**Comment by mckaygerhard** [comment]
take easy.. i''ll try a special test case to send to you.. i'm some busy but i'll provide the problem case.. postgres are not so fatal.. works in many cases...

fatal cases are with odbc.. mostly by two problems:
- cursors are forward only
- connections must be closes to make other query if a open cursor are opened. (previous query)

so the component are not so "production" due only works 100% with mysql in mayor cases and 90% with sqlite and postgres.. 

currently i'm waiting by at leas the second problem in gambas bugtracker.. but for make some help to you i need analize deeply and understand your code more..

**Comment by chthal** [comment]
Updated Readme. Feel free to fork it and make it compatible with other databases/connections.

**Comment by mckaygerhard** [comment]
i see it, when some progress are on a similar control for data, i send a project with docs

**Comment by chthal** [comment]
Ok, then happy programming. This thingy here cost about a half a year of development (and fixing). And it still contains some small bugs ... :-)

**Comment by mckaygerhard** [comment]
hi, yeah its not easy.. i based the work in that made by shordi in gambas-es, of course shordi does not have some vision and think only exist sqlite and mysql worlds..  

now i can see that your code are very similar like that: https://www.gambas-es.org/viewtopic.php?t=6455&p=38986#p38986 the upload of their code are here: https://www.gambas-es.org/viewtopic.php?f=4&t=6455&start=0 

their Working like the flintstone era, with loads that replacing the previous tarball, you can not follow the changes regarding the old code, so track their change cost to me too much.. and i not expert..

the principe its loads all the cursor result in ram, then here made all the filters, searchs, and if user want something "inverse" (example "not equal to" event "equal to") provide a inverse in the sorted search.. 

this principe seem are partially done in the component made by me in gitlab, but currently uses the Result object.. so firts we need traspasing the Result (that are forward only) to a Variant bidimensional object and then made the operation over the Variant object
